### ISSUE TITLE
MX decoder again

### ISSUE THREAD

**Issue by KP1533TM2** [davidgiven/fluxengine#466]
Hello David.
I'm not sure how I overlooked it the last time, sorry - I might've used my pull request version to decode some of my disks.
Hope I'm not too much of a bother.
New MX decoder generally works, but I noticed now that fresh fluxengine I just checked out is generating 400K, not 440K MX images again.
I've looked into it, seems it's a simple one-off error in the sector count:

**arch/mx/decoder.cc, line 41:**
`if (_currentSector == 10)` - should be 11, otherwise all 10th sectors get ignored.

Regarding MX blanks: I don't have truly blank MX disks, as it turned out. Would a synthetic flux image (along with an *.img that checks against it) suffice for a test suite, at least for the time being?
(I figured out enough DEC RT-11 OS to initialize a disk image with native software on an emulator)

Best regards,
Alexei

**Comment by davidgiven** [comment]
Bah, that was stupid. Fixing, thanks.

Re the MX blanks: currently I'm using synthetic MX images as part of the unit tests (I make an flux image, then decode it again and ensure that I get back the same data), which is fine for quick tests but I'd rather like something made by real hardware to test with. However, thinking about it, if there's a disk that contains non-copyrighted stuff then that will probably do fine if you have one...

**Comment by KP1533TM2** [comment]
[test_image.zip](https://github.com/davidgiven/fluxengine/files/8125126/test_image.zip)
I've included disk image I've previously talked about in a pull request a week ago, now in format as they're kept in your `fluxengine-testdata` repo.
I've also included all files extracted from this image and converted to readable UTF8.

- SPEC.txt.txt - [this](https://zxpress.ru/book.php?id=119) book. Old, self-published (apparently), guide to ZX Spectrum.
- MONZXS.txm.txt - [this](https://zxpress.ru/book_articles.php?id=128) part of [this](https://zxpress.ru/book.php?id=57) book. Some ZX Spectrum ROM info.
- P3.mac.txt - looks like some kind of PDP-11 (1806ВМ2) assembly boot code; can't find anything about it, there's no copyright or credit notice.

Most of my other disks, unfortunately, either contain an OS, some parts of it, Spectrum/clone ROMs, work documents etc.
